### PR TITLE
Refactor Dockerfiles for development and deployment

### PR DESCRIPTION
**Pull request by yakloinsteak**
## Description

Refactor and simplify the dockerfiles. Removes the possibility of stale application files. 

Clear cache by going to github actions -> Cache and deleting the cache for this build (Linux-buildx-warehouse)

## Manual steps

See testing steps. Do a build and then fix your permissions

## How to test

- `docker compose build web`
- `docker compose run -u 0 shell chown -R app-user:app-user /bundle /app /node_modules`
-  Run a web server and shell with the newly built image. Make sure `bundle`, `rails c`, `rake`, etc. all still work
-  Deploy to at least one environment 

## Type of change

- [x] Bug fix
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: docker/app/entrypoint.sh [hunk 1]
```diff
@@ -46,17 +46,21 @@ T1=`date +%s`
 ASSET_CHECKSUM=$(ASSETS_PREFIX=${ASSETS_PREFIX} ./bin/asset_checksum) # This should return the same hash as the call in asset_compiler.rb
 T2=`date +%s`
 echo "...checksumming took $(expr $T2 - $T1) seconds"
+
+echo "asset.checksum.log"
+cat asset.checksum.log
```
Comment (eanders) [inline docker/app/entrypoint.sh]: Did you intend to leave these in?

Comment (yakloinsteak) [reply docker/app/entrypoint.sh]: No. I'll remove them or comment them out.

**Comment by eanders** [comment]
Need to login to the shell console, then change ownership before this will work.
```
docker-compose run -u 0 shell chown -R app-user:app-user /bundle /app /node_modules
```

**Comment by eanders** [comment]
@yakloinsteak things seem to have proceeded normally overnight.  I'm going to merge shortly, unless you have objections.

**Review by eanders — APPROVED** [review]

**Comment by yakloinsteak** [comment]
Go for it. Since last night was normal, I feel pretty good I didn't miss anything critical.

File: config/deploy/docker/assets/Dockerfile.open-path-warehouse.pre-cache [hunk 2]
```diff
@@ -1,112 +0,0 @@
...
-
-COPY Gemfile Gemfile.lock Rakefile config.ru package.json yarn.lock ./
-COPY bin ./bin
-COPY public ./public
-COPY lib ./lib
-COPY config ./config
-COPY app ./app
-COPY vendor ./vendor
-COPY config/deploy/docker/assets/dotenv.temporary ./.env
-COPY sample.irbrc ./.irbrc
```
Comment (eanders) [inline config/deploy/docker/assets/Dockerfile.open-path-warehouse.pre-cache:37]: @yakloinsteak Looks like this got missed as well.  Any thoughts on where best to put this?

Comment (yakloinsteak) [reply config/deploy/docker/assets/Dockerfile.open-path-warehouse.pre-cache:37]: https://github.com/greenriver/hmis-warehouse/pull/4004 Do we need the .env too?

Comment (eanders) [reply config/deploy/docker/assets/Dockerfile.open-path-warehouse.pre-cache:37]: Not as far as I can tell.  I think that had something to do with assets maybe?

**Pull request merged**